### PR TITLE
Named function arguments

### LINKED ISSUE #22: Solidity feature: Named Arguments (in function calls)
E.g. `f({value: 2, key: 3});`

### PR DESCRIPTION
**Pull request by dfilaretti**
Fixes #22. 
Adds support for named function arguments. 
Please see comment below about regressions.

### CODE REVIEW

**Comment by dfilaretti** [comment]
In addition to the new passing tests, we have a regression:

Test `inconsistency` (from `SolidityOptimizer.cpp`) was previously not compiling, and now has 3 failing assertions. 

* `Optimizer did not reduce bytecode size. Non-optimized size: 2115 - optimized size: 2115`
* `SolidityOptimizer.cpp(100): error in "inconsistency": No optimized output for trigger()`
* `SolidityOptimizer.cpp(101): error in "inconsistency": No un-optimized output for trigger()`

See #137

File: libsolidity/codegen/IeleCompiler.cpp [hunk 1]
```diff
@@ -2354,13 +2354,42 @@ bool IeleCompiler::visit(const FunctionCall &functionCall) {
     return false;
   }
 
-  if (!functionCall.names().empty()) {
-    solAssert(false, "not implemented yet");
-    return false;
+  std::vector<ASTPointer<const Expression>> arguments;
+  FunctionTypePointer functionType;
+	auto callArgumentNames = functionCall.names();
```
Comment (dwightguth) [inline libsolidity/codegen/IeleCompiler.cpp]: Looks like your indentation is screwed up here

File: libsolidity/codegen/IeleCompiler.cpp [hunk 2]
```diff
@@ -2354,13 +2354,42 @@ bool IeleCompiler::visit(const FunctionCall &functionCall) {
...
+          *functionCall.expression().annotation().type);
+    const StructType &structType =
+      dynamic_cast<const StructType &>(*type.actualType());
+    functionType = structType.constructorType();
+
+    solAssert(functionType->parameterTypes().size() ==
+        structType.memoryMemberTypes().size(),
+        "IeleCompiler: struct constructor with missing arguments");
+  }
+  else {
```
Comment (dwightguth) [inline libsolidity/codegen/IeleCompiler.cpp]: This should be on the same line as the prior closing brace

File: libsolidity/codegen/IeleCompiler.cpp [hunk 3]
```diff
@@ -2354,13 +2354,42 @@ bool IeleCompiler::visit(const FunctionCall &functionCall) {
...
+		for (auto const& parameterName: functionType->parameterNames())
+		{
+			bool found = false;
+			for (size_t j = 0; j < callArgumentNames.size() && !found; j++)
+				if ((found = (parameterName == *callArgumentNames[j])))
+					arguments.push_back(callArguments[j]);
+			solAssert(found, "");
+		}
+  }
+  else {
```
Comment (dwightguth) [inline libsolidity/codegen/IeleCompiler.cpp]: Same here

File: libsolidity/codegen/IeleCompiler.cpp [hunk 4]
```diff
@@ -2354,13 +2354,42 @@ bool IeleCompiler::visit(const FunctionCall &functionCall) {
...
+    functionType = std::dynamic_pointer_cast<const FunctionType>(
+        functionCall.expression().annotation().type);
   }
 
-  const std::vector<ASTPointer<const Expression>> &arguments =
-    functionCall.arguments();
+  if (!callArgumentNames.empty()) {
+		for (auto const& parameterName: functionType->parameterNames())
```
Comment (dwightguth) [inline libsolidity/codegen/IeleCompiler.cpp]: Indentation seems too much here

**Pull request merged**